### PR TITLE
Use local lookup policy for generated image stream resources

### PR DESCRIPTION
**Pull request by iocanel**
When using kubernetes resources in Openshift we run into the issue of needing to now the project name upfront.
This is because the project name influences the image repository.

A solution to this issue is described here: https://access.redhat.com/documentation/en-us/openshift_container_platform/4.8/html/images/using-imagestreams-with-kube-resources

The pull request applies the first of the two approaches described (using local lookup policy for images).

### CODE REVIEW

File: extensions/kubernetes/vanilla/deployment/src/main/java/io/quarkus/kubernetes/deployment/OpenshiftConfig.java [hunk 1]
```diff
@@ -78,6 +78,12 @@ public static enum DeploymentResourceKind {
     @ConfigItem
     Optional<String> partOf;
 
+    /**
+     * When this property is set to true, on deployment time the name of the project will be set as image group.
+     */
+    @ConfigItem(defaultValue = "true")
+    boolean projectAsImageGroup;
+
```
Comment (Sgitario) [inline extensions/kubernetes/vanilla/deployment/src/main/java/io/quarkus/kubernetes/deployment/OpenshiftConfig.java:86]: I think this is a leftover

**Review by Sgitario — COMMENTED** [review]
These changes look good to me. Only one leftover spotted.

**Review by Sgitario — APPROVED** [review]
lgtm

**Comment by gsmet** [comment]
CI is failing because of this.

**Comment by iocanel** [comment]
Fixed broken test.

**Pull request merged**